pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -100,6 +103,136 @@ def test_rpm(self):
+        self.assertIsNotNone(repo['_latest_version_href'])
+
+        self.assertDictEqual(
+            get_content_summary(repo),
+            RPM_KICKSTART_FIXTURE_SUMMARY
+        )
+        self.assertDictEqual(
+            get_added_content_summary(repo),
+            RPM_KICKSTART_FIXTURE_SUMMARY
+        )
+
+        # Sync the repository again.
+        latest_version_href = repo['_latest_version_href']
+        sync(self.cfg, remote, repo)
+        repo = self.client.get(repo['_href'])
+
+        # Check that nothing has changed since the last sync.
+        self.assertNotEqual(latest_version_href, repo['_latest_version_href'])
+        self.assertDictEqual(
+            get_content_summary(repo),
+            RPM_KICKSTART_FIXTURE_SUMMARY
+        )
+        self.assertDictEqual(get_added_content_summary(repo), {})
+
+    def test_rpm_kickstart_on_demand(self):
+        """Sync repositories with the rpm plugin.
+
+        This test targets the following issue:
+
+
+        `Pulp #5202 <https://pulp.plan.io/issues/5202>`_
+
+        In order to sync a repository a remote has to be associated within
+        this repository. When a repository is created this version field is set
+        as None. After a sync the repository version is updated.
+
+        Do the following:
+
+        1. Create a repository and a remote.
+        2. Assert that repository version is None.
+        3. Sync the remote.
+        4. Assert that repository version is not None.
+        5. Assert that the correct number of units were added and are present
+           in the repo.
+        6. Sync the remote one more time.
+        7. Assert that repository version is different from the previous one.
+        8. Assert that the same number of are present and that no units were
+           added.
+        """
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+
+        # Create a remote with the standard test fixture url.
+        body = gen_rpm_remote(url=RPM_KICKSTART_FIXTURE_URL, policy='on_demand')
+        remote = self.client.post(RPM_REMOTE_PATH, body)
+        self.addCleanup(self.client.delete, remote['_href'])
+
+        # Sync the repository.
+        self.assertIsNone(repo['_latest_version_href'])
+        sync(self.cfg, remote, repo)
+        repo = self.client.get(repo['_href'])
+        for kickstart_content in get_content(repo)[RPM_KICKSTART_CONTENT_NAME]:
+            self.addCleanup(self.client.delete, kickstart_content['_href'])
+
+        # Check that we have the correct content counts.
+        self.assertIsNotNone(repo['_latest_version_href'])
+
+        self.assertDictEqual(
+            get_content_summary(repo),
+            RPM_KICKSTART_FIXTURE_SUMMARY
+        )
+        self.assertDictEqual(
+            get_added_content_summary(repo),
+            RPM_KICKSTART_FIXTURE_SUMMARY
+        )
+
+        # Sync the repository again.
```

Comment:
I don't see additional value in this re-sync aspect of this test given that it's in other tests already.

What is important for on_demand correctness assertion is that the content that is available "on_demand" can actually be fetched by Pulp smash. Can you assert that additionally via `get_content()` calls and validating the checksums of the content you are recieving. This would simulate a client actually fetching it which is key.